plugins/outputs/websocket/websocket.go

```diff
@@ -0,0 +1,206 @@
+package websocket
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+
+	ws "github.com/gorilla/websocket"
+)
+
+var sampleConfig = `
+  ## URL is the address to send metrics to. Make sure ws or wss scheme is used.
+  url = "ws://127.0.0.1:8080/telegraf"
+
+  ## Timeouts (make sure read_timeout is larger than server ping interval or set to zero).
+  # connect_timeout = "5s"
+  # write_timeout = "5s"
+  # read_timeout = "30s"
+
+  ## Optionally turn on using text data frames (binary by default).
+  # use_text_frames = false
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  # data_format = "influx"
+
+  ## Additional HTTP Upgrade headers
+  # [outputs.websocket.headers]
+  #   Authorization = "Bearer <TOKEN>"
+`
+
+const (
+	defaultConnectTimeout = 5 * time.Second
+	defaultWriteTimeout   = 5 * time.Second
+	defaultReadTimeout    = 30 * time.Second
+)
+
+// WebSocket can output to WebSocket endpoint.
+type WebSocket struct {
+	URL            string            `toml:"url"`
+	ConnectTimeout config.Duration   `toml:"connect_timeout"`
+	WriteTimeout   config.Duration   `toml:"write_timeout"`
+	ReadTimeout    config.Duration   `toml:"read_timeout"`
+	Headers        map[string]string `toml:"headers"`
+	UseTextFrames  bool              `toml:"use_text_frames"`
+	tls.ClientConfig
+
+	conn       *ws.Conn
+	serializer serializers.Serializer
+}
```

Comment:
Add the logger to use it in the read function, in tests set it to testutils.Logger{}. Telegraf will set it automatically at runtime
```suggestion
	serializer serializers.Serializer
	Logger     telegraf.Logger `toml:"-"`
}
```

Reply:
Thanks, done 👍